arangod/RestServer/LanguageCheckFeature.cpp

```diff
@@ -37,89 +37,137 @@
 
 namespace {
 
+constexpr std::string_view defaultLangKey = "default";
+constexpr std::string_view icuLangKey = "icu-language";
+
 /// @brief reads previous default langauge from file
 arangodb::Result readLanguage(arangodb::ArangodServer& server,
-                              std::string& language) {
+                              std::string& defaultLanguage,
+                              std::string& icuLanguage) {
   auto& databasePath = server.getFeature<arangodb::DatabasePathFeature>();
   std::string filename = databasePath.subdirectoryName("LANGUAGE");
 
   if (!TRI_ExistsFile(filename.c_str())) {
     return TRI_ERROR_FILE_NOT_FOUND;
   }
 
-  std::string found;
   try {
     VPackBuilder builder =
         arangodb::basics::VelocyPackHelper::velocyPackFromFile(filename);
     VPackSlice content = builder.slice();
     if (!content.isObject()) {
       return TRI_ERROR_INTERNAL;
     }
-    VPackSlice defaultSlice = content.get("default");
-    if (!defaultSlice.isString()) {
+    VPackSlice defaultSlice = content.get(defaultLangKey);
+    VPackSlice icuSlice = content.get(icuLangKey);
+
+    // both languages are specified in files
+    if (defaultSlice.isString() && icuSlice.isString()) {
+      LOG_TOPIC("4fa52", ERR, arangodb::Logger::CONFIG)
+          << "Only one language should be specified";
+      return TRI_ERROR_INTERNAL;
+    }
+
+    if (defaultSlice.isString()) {
+      defaultLanguage = defaultSlice.stringView();
+    } else if (icuSlice.isString()) {
+      icuLanguage = icuSlice.stringView();
+    } else {
       return TRI_ERROR_INTERNAL;
     }
-    found = defaultSlice.copyString();
+
   } catch (...) {
     // Nothing to free
     return TRI_ERROR_INTERNAL;
   }
 
-  language = found;
-  LOG_TOPIC("c499e", TRACE, arangodb::Logger::CONFIG)
-      << "using default language: " << found;
+  if (!defaultLanguage.empty()) {
+    LOG_TOPIC("c499e", TRACE, arangodb::Logger::CONFIG)
+        << "using default language: " << defaultLanguage;
+  } else {
+    LOG_TOPIC("c490e", TRACE, arangodb::Logger::CONFIG)
+        << "using icu language: " << icuLanguage;
+  }
```

Comment:
What if both languages are empty strings  e,g, we have the following LANGUAGE file  `{icu-language:""}`